src/Planners/src/UnicyclePlanner.cpp

```diff
@@ -0,0 +1,600 @@
+
+    if (!m_pImpl->planner)
+    {
+        log()->error("{} The Unicycle planner has never been initialized.", logPrefix);
+        return false;
+    }
+
+    m_pImpl->output = std::nullopt;
+
+    // Initialize the left FootPrint
+    m_pImpl->left = std::make_shared<FootPrint>();
+    m_pImpl->left->setFootName(m_pImpl->names.left);
+
+    // Initialize the right FootPrint
+    m_pImpl->right = std::make_shared<FootPrint>();
+    m_pImpl->right->setFootName(m_pImpl->names.right);
+
+    // Compute the FootPrints
+    if (!m_pImpl->planner->computeNewSteps(m_pImpl->left,
+                                           m_pImpl->right,
+                                           m_pImpl->horizon.t0,
+                                           m_pImpl->horizon.tf))
+    {
+        m_pImpl->left = nullptr;
+        m_pImpl->right = nullptr;
+        log()->error("{} Failed to compute new steps.", logPrefix);
+        return false;
+    }
+
+    // Create and configure the generator
+    auto generator = UnicycleGenerator();
+    generator.setSwitchOverSwingRatio(m_pImpl->gait.stancePhaseRatio);
+    double lastStepSwitchTime = 0.8;
+    generator.setTerminalHalfSwitchTime(lastStepSwitchTime);
+    generator.setPauseConditions(m_pImpl->duration.max, m_pImpl->duration.nominal);
+
+    // Compute the contact states using the generator
+    if (!generator.generateFromFootPrints(m_pImpl->left,
+                                          m_pImpl->right,
+                                          m_pImpl->horizon.t0,
+                                          m_pImpl->dt.planner))
+    {
+        log()->error("{} Failed to generate from footprints.", logPrefix);
+        return false;
+    }
+
+    // Get the contact states over the horizon
+    std::vector<bool> leftStandingPeriod;
+    std::vector<bool> rightStandingPeriod;
+    generator.getFeetStandingPeriods(leftStandingPeriod, rightStandingPeriod);
+
+    // Lambda to convert the vector with contact states to a vector of PlannedContact.
+    // It just processes timings, we fill the transform later.
+    auto processStandingPeriod
+        = [](const std::vector<bool>& isFootInContactVector,
+             const double dt,
+             const std::string& contactName) -> std::vector<Contacts::PlannedContact> {
+        std::vector<Contacts::PlannedContact> contacts;
+
+        // Loop over all the horizon
+        for (size_t i = 1; i < isFootInContactVector.size(); ++i)
+        {
+            const bool thisState = isFootInContactVector[i];
+            const bool lastState = isFootInContactVector[i - 1];
+            const bool aboutToLand = lastState == 0 && thisState == 1;
+
+            // Create a new contact
+            if (i == 1 || aboutToLand)
+            {
+                auto contact = Contacts::PlannedContact();
+                contact.name = contactName;
+                contact.activationTime = dt * i;
+                contact.deactivationTime = contact.activationTime;
```

Comment:
Since the activation time could be before the beginning of the horizon, I would initialize given the corresponding time instant (i.e. t0 + dt * i).

Reply:
Let's see how we agree on https://github.com/dic-iit/bipedal-locomotion-framework/pull/320/files/896fe514f680bd9ef6fc140350ee7f0afcdfaad2#r684128023 first. Then, the `+ dt` is already performed below (also for the first step) at line 542.